src/internal.h

```diff
@@ -242,6 +246,34 @@ DISPATCH_EXPORT DISPATCH_NOTHROW void dispatch_atfork_child(void);
 #include <unistd.h>
 #endif
 
+#if USE_POSIX_SEM
+#include <semaphore.h>
+#elif USE_FUTEX_SEM
+#include <sys/syscall.h>
+#include <linux/futex.h>
+#endif
+
+#if defined(__APPLE__)
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+#define DISPATCH_LITTLE_ENDIAN
+#elif __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+#define DISPATCH_BIG_ENDIAN
+#endif
+
+#elif defined(__linux__)
+#include <endian.h>
+#if __BYTE_ORDER == __LITTLE_ENDIAN
+#define DISPATCH_LITTLE_ENDIAN
+#elif __BYTE_ORDER == __BIG_ENDIAN
+#define DISPATCH_BIG_ENDIAN
+#endif
+
+#endif  /* defined(__APPLE__) */
+
+#if !(defined(DISPATCH_BIG_ENDIAN) || defined(DISPATCH_LITTLE_ENDIAN))
```

Comment:
confusing, proper way to do that is let the compiler do it.
I don't know whether APPLE does it that way but properway is:
#if defined(__APPLE__)
#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
#define DISPATCH_LITTLE_ENDIAN
#elif __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
#define DISPATCH_BIG_ENDIAN
#endif
#else 
#include <endian.h>
#if __BYTE_ORDER == __LITTLE_ENDIAN
#define DISPATCH_LITTLE_ENDIAN
#elif __BYTE_ORDER == __BIG_ENDIAN
#define DISPATCH_BIG_ENDIAN
#else
#error "please define DISPATCH_LITTLE_ENDIAN or DISPATCH_BIG_ENDIAN for your
architecture / platform"
#endif  /* defined(__APPLE__) */